atomic_reactor/util.py

```diff
@@ -320,7 +323,12 @@ def clone_git_repo(git_url, target_dir, commit=None, retry_times=GIT_MAX_RETRIES
     logger.debug("url = '%s', dir = '%s', commit = '%s'",
                  git_url, target_dir, commit)
 
-    cmd = ["git", "clone", git_url, quote(target_dir)]
+    cmd = ["git", "clone"]
+    if branch:
+        cmd += ["-b", branch, "--shallow-branch"]
+    if depth:
+        cmd += ["--depth", '{0}'.format(depth)]
+    cmd += [git_url, quote(target_dir)]
 
     logger.debug("cloning '%s'", cmd)
     for counter in range(retry_times + 1):
```

Comment:
I think we need to make sure that if we fail to reset to the expected commit ID the first time, we `git fetch --unshallow` and try again. Nothing prevents other commits being added to the branch in between the orchestrator build measuring the commit depth and the worker build (here) cloning it again.

Reply:
Thanks, added comments on specific parts of it.